src/utils/ddb/ddb_main.c

```diff
@@ -180,11 +180,35 @@ process_line_cb(void *cb_args, char *line, uint32_t line_len)
 	/* ignore empty lines */
 	if (all_whitespace(line, line_len))
 		return 0;
-	return ddb_run_cmd(ctx, line, ctx->dc_write_mode);
+	return ddb_run_cmd(ctx, line);
 }
 
 #define str_has_value(str) ((str) != NULL && strlen(str) > 0)
 
+static int
+open_if_needed(struct ddb_ctx *ctx, struct program_args *pa, bool *open)
+{
+	int rc = 0;
+
+	if (!str_has_value(pa->pa_pool_path)) {
+		*open = false;
+		goto out;
+	}
+	ctx->dc_pool_path = pa->pa_pool_path;
+
+	if (str_has_value(pa->pa_r_cmd_run)) {
+		rc = ddb_parse_cmd_str(ctx, pa->pa_r_cmd_run, open);
+		if (rc)
+			return rc;
+	}
+
+	if (str_has_value(pa->pa_cmd_file))
+		*open = true;
+
+out:
+	return rc;
+}
+
 int
 ddb_main(struct ddb_io_ft *io_ft, int argc, char *argv[])
```

Comment:
A little confused because I can't figure out what `ddb_main` is used for. It seems to think it is the main function of ddb, but the Go stuff is the entry point now. No references to `ddb_main` outside of this directory. Should it (and the superfluous `main` function in ddb_entry.c) be removed?

Reply:
Yes, i am also a bit confused about not sure about this, i think maybe we could remove it, but i'd like to keep it is for now, and remove it in the separate discussion if you are fine with this.

Reply:
OK. I won't block either way. It would be great to prune unused code whenever we can.